kratos/processes/calculate_embedded_nodal_variable_from_skin_process.h

```diff
@@ -398,7 +506,7 @@ class KRATOS_API(KRATOS_CORE) CalculateEmbeddedNodalVariableFromSkinProcess : pu
         #pragma omp parallel for
         for (int i_node = 0; i_node < static_cast<int>(r_int_elems_model_part.NumberOfNodes()); ++i_node) {
             const auto it_node = r_int_elems_model_part.NodesBegin() + i_node;
-            auto &r_emb_nod_val = (mrBaseModelPart.GetNode(it_node->Id())).FastGetSolutionStepValue(mrEmbeddedNodalVariable);
+            auto &r_emb_nod_val = (mrBaseModelPart.GetNode(it_node->Id())).FastGetSolutionStepValue(mrEmbeddedNodalVariable, mBufferPosition);
```

Comment:
does the base modelpart have exactly the same nodes as the modelpart you iterate on?

because if it is so, you can avoid the access by id (which is logn) in favor of constant access time

Reply:
No. The model part I am iterating contains those nodes belonging to intersected edges. I preserve the same ids. when creating the model part but the number depends on the intersection patterns.

Reply:
ok